dpctl/tensor/libtensor/include/utils/sycl_complex.hpp

```diff
@@ -26,11 +26,25 @@
 
 #pragma once
 
-#define SYCL_EXT_ONEAPI_COMPLEX
+#ifndef SYCL_EXT_ONEAPI_COMPLEX
+#define SYCL_EXT_ONEAPI_COMPLEX 1
+#endif
 #if __has_include(<sycl/ext/oneapi/experimental/sycl_complex.hpp>)
 #include <sycl/ext/oneapi/experimental/sycl_complex.hpp>
 #else
 #include <sycl/ext/oneapi/experimental/complex/complex.hpp>
 #endif
 
-namespace exprm_ns = sycl::ext::oneapi::experimental;
+namespace dpctl
+{
+namespace tensor
+{
+namespace sycl_utils
+{
+
+template <typename T>
+using sycl_complex_t = sycl::ext::oneapi::experimental::complex<T>;
```

Comment:
It sounds like we have to use `sycl_complex_t` type everywhere to have it working with HIP backend.
In that case would it be worse idea to cast `usm_ndarray` to `sycl_complex_t` type from the beginning (or at least in case of building with HIP support enabled)? It might be that `sycl_complex_t = std::complex<T>` by default.

It would impact all type matrix where we need to replace `std::complex` with `sycl_complex_t` and probably to update some helper functions and aliases. But that would help us avoid casting to `sycl_complex_t` every time we need to do some math operation.

Reply:
I see, clear now. It makes a lot of sense.